packages/verkle/src/verkleTree.ts

```diff
@@ -213,11 +215,169 @@ export class VerkleTree {
    * @param value - the value to store
    * @returns A Promise that resolves once value is stored.
    */
-  // TODO: Rewrite following logic in verkle.spec.ts "findPath validation" test
-  async put(_key: Uint8Array, _value: Uint8Array): Promise<void> {
-    throw new Error('not implemented')
+  async put(key: Uint8Array, value: Uint8Array): Promise<void> {
+    if (key.length !== 32) throw new Error(`expected key with length 32; got ${key.length}`)
+    const stem = key.slice(0, 31)
+    const suffix = key[key.length - 1]
+    this.DEBUG && this.debug(`Stem: ${bytesToHex(stem)}; Suffix: ${suffix}`, ['PUT'])
+
+    const putStack: [Uint8Array, VerkleNode][] = []
+    // Find path to nearest node
+    const foundPath = await this.findPath(stem)
+
+    // Sanity check - we should at least get the root node back
+    if (foundPath.stack.length === 0) {
+      throw new Error(`Root node not found in trie`)
+    }
+
+    // Step 1) Create or update the leaf node
+    let leafNode: LeafNode
+    // First see if leaf node already exists
+    if (foundPath.node !== null) {
+      // Sanity check to verify we have the right node type
+      if (foundPath.node.type !== VerkleNodeType.Leaf) {
+        throw new Error(
+          `expected leaf node found at ${bytesToHex(stem)}. Got internal node instead`
+        )
+      }
+      leafNode = foundPath.node as LeafNode
```

Comment:
I can create some type guards for those, I think this will be useful in qute a few places.

Reply:
Just did: [`9b00803` (#3473)](https://github.com/ethereumjs/ethereumjs-monorepo/pull/3473/commits/9b00803c1708ce30839ac2bfd28be87cef768c00)

Reply:
Sounds good!